### ISSUE TITLE
Inconsistent assumptions over interface React

### ISSUE THREAD

**Issue by b0o** [bloodyowl/reshowcase#38]
I am trying to set up the reshowcase [demo](https://github.com/bloodyowl/reshowcase/blob/main/example/Demo.res) in my project and getting the following error:

```
rescript: [1/1] src/showcase/Demo.cmj
FAILED: src/showcase/Demo.cmj

  We've found a bug for you!
  /path/to/my-project/src/showcase/Demo.res

  It's possible that your build is stale.
  Try to clean the artifacts and build again?

  Here's the original error message
  The files /path/to/my-project/node_modules/reshowcase/lib/ocaml/Entry-Reshowcase.cmi
  and /path/to/my-project/node_modules/@rescript/react/lib/ocaml/React.cmi
  make inconsistent assumptions over interface React

FAILED: cannot make progress due to previous errors.
>>>> Finish compiling(exit: 1)
```

Based on my [limited research](https://forum.rescript-lang.org/t/inconsistent-assumptions-upgrading-to-rescript-react-vs-graphql-ppx/2077/4) I believe this is because reshowcase is using `reason-react` while my project is using `@rescript/react`. 

Side note: The example code in the README doesn't work at all, it seems the demo function signature has changed but that snipped wasn't updated?

This project looks really cool and I'm excited to try it out!

**Comment by bloodyowl** [comment]
Hi, thanks for the report.

I think you're right. I'll release an update this week.

**Comment by bloodyowl** [comment]
Released as 5.0.0!

**Comment by b0o** [comment]
It's working! Thank you.
